src/bun.js/api/BunObject.zig

```diff
@@ -3397,31 +3394,39 @@ pub fn serve(
                         },
                     };
 
-                    var server = ServerType.init(config, globalObject);
+                    const server = ServerType.init(config, globalObject) catch |e| {
+                        config.deinit();
+                        return e;
+                    };
+
                     if (globalObject.hasException()) {
```

Comment:
can we delete this check? does the error thrown from `ServerType.init` cover it?

Reply:
no 😭

Reply:
i will audit locations like this soon in my project